AmplifyPlugins/Analytics/AWSPinpointAnalyticsPlugin/LocalStorage/AnalyticEventStorage.swift

```diff
@@ -0,0 +1,95 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import SQLite
```

Comment:
shouldn't the `AnalyticsEventStorage` be storage-agnostic? If so, the logic specific to a SQLite adapter (i.e. how to retrieve the actual data from the storage) should be part of `SQLiteLocalStorageAdapter` only.

Reply:
I think not having `AnalyticsEventStorage` tied to a specific implementation of a storage will simplify testing of its consumers

Reply:
I think it's not made agnostic because this class is the replacement for `AWSPinpointEventRecorder`, which included all the SQL queries within. 

But maybe we can improve on that, but eventually we'll need to have the SQL-queries being generated somewhere. Perhaps inside `SQLiteLocalStorageAdapter` ?

Reply:
The initial intent of AnalyticsEventStorage is just a placeholder class for now to hold all of the SQL queries. Based on the feedback, I've made an AnalyticsEventStorage protocol that is storage agnostic and an AnalyticsEventsSQLStorage implementation that is SQL specific.  All SQL queries and create statements should be in this SQL implimentation, I'd imagine that this is where all Pinpoint Events operations gets converted to SQL.  The SQLiteLocalStorageAdapter becomes a thin lightweight interface to the database, and all it does it executes SQL commands.  Let me know if that is a workable solution.